strip backticks around procedure completion items, correct insertion …

HerrEmil opened the pull request:
also corrects the insertion range of procedures

Before change:
![old](https://user-images.githubusercontent.com/939458/109161570-815b7400-7777-11eb-94e8-f8bf9c6e479b.gif)

After change:
![new](https://user-images.githubusercontent.com/939458/109161603-8b7d7280-7777-11eb-932f-53d42693adbf.gif)

Demo: http://strip-backticks.surge.sh/

OskarDamkjaer commented on src/shared/modules/editor/editorDuck.ts:
```diff
@@ -139,11 +161,19 @@ export const populateEditorFromUrlEpic = (some$: any, _store: any) => {
 }
 
+// CypherEditorSupport returns the content attributes of procedures with dots wrapped in backticks, e.g. "`apoc.coll.avg`"
+// This function strips any surrounding backticks before we use the .content value in the completion item provider
```
👍

OskarDamkjaer commented on src/shared/modules/editor/editorDuck.ts:
```diff
@@ -261,16 +295,14 @@ export const updateEditorSupportSchemaEpic: Epic<Action, any> = (
-        ] as string[],
-        procedures: state.meta.procedures.map(toProcedure) as string[]
+        ],
+        procedures: state.meta.procedures.map(toProcedure)
```
Nice to see these ts improvements!

oskarhane reviewed (commented):
Backticks enables tokens (not sure what to call them, but labels, prop-types, procedure names etc) to contain characters like `-`.

This would fail: 
```
CREATE (:Kor-v)
```

This would pass: 
```
CREATE (:`Kor-v`)
```

The problem with this change is that nothing will be escaped anymore.
If you create the above label and type a `MATCH` clause to autocomplete, it will autocomplete with `MATCH (n:Kor-v) RETURN n` which would fail. 

So we do want to keep the back-ticks. Is there a way to work around this?
@HerrEmil @OskarDamkjaer

OskarDamkjaer commented:
@oskarhane We do have the completion-type available, is there an issue with only stripping backticks for procedures?

HerrEmil commented:
@oskarhane @OskarDamkjaer I added a check for completion type and only strip surrounding backticks on functions and procedures. I've updated the surge link which works with \`Kor-v\` example now

oskarhane reviewed (approved):
Wohoo, that work better!

Can we write some tests to show that this is intended and that we don't break it in the future?

HerrEmil commented:
Added a few unit tests and plan on merging on green test builds :)

OskarDamkjaer reviewed (approved):
:+1:
